tests/execution.rs

```diff
@@ -3435,6 +3453,69 @@ fn test_invalid_exit_or_return() {
     }
 }
 
+#[test]
+fn callx_unsupported_instruction_and_exceeded_max_instructions() {
+    test_interpreter_and_jit_asm!(
+        "
+        sub32 r7, r1
+        sub64 r5, 8
+        sub64 r7, 0
+        callx r5
+        callx r5
+        return
+        ",
+        [],
+        TestContextObject::new(4),
+        ProgramResult::Err(EbpfError::UnsupportedInstruction),
+    );
+
+    let prog = &[
+        0x1c, 0x17, 0x17, 0x95, 0x61, 0x61, 0x61, 0x61, // sub32
+        0x17, 0x55, 0x54, 0xff, 0x08, 0x00, 0x00, 0x00, // sub64
+        0x17, 0x17, 0x17, 0xff, 0x00, 0x00, 0x00, 0x00, // sub64
+        0x8d, 0x55, 0x54, 0xff, 0x09, 0x00, 0x00, 0x00, // callx
+        0x8d, 0x55, 0x54, 0x23, 0x08, 0x00, 0x11, 0x4e, // callx
+    ];
+
+    let loader = Arc::new(BuiltinProgram::new_loader(
+        Config::default(),
+        FunctionRegistry::default(),
+    ));
+    let mut executable = Executable::<TestContextObject>::from_text_bytes(
+        prog,
+        loader,
+        SBPFVersion::V2,
+        FunctionRegistry::default(),
+    )
+    .unwrap();
+
+    test_interpreter_and_jit!(
+        true,
+        false,
+        executable,
+        [],
+        TestContextObject::new(4),
+        ProgramResult::Err(EbpfError::UnsupportedInstruction)
+    );
+}
+
+#[test]
+fn test_maximum_after_callx() {
```

Comment:
How is this different from the second part of `test_err_capped_before_exception()`?

Reply:
The callx destination is valid here.

Reply:
Alright, how about renaming "maximum" to "capped" like the other tests?